### ISSUE TITLE
'make install' to wrong directory

### ISSUE THREAD

**Issue by dbeeman** [borismarin/genesis2.4gamma#19]
The Makefile generated by configure should copy files from src to '../'. but does not.

**Comment by physicalist** [comment]
The 'make install' in the autoconf case actually works as intended by the GNU people, as @prefix@ [defaults](https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/Default-Prefix.html) to `/usr/local` if configure is not run as `./configure --prefix=<path>`.

To prevent unexpected behaviour, particularly for long-time users of GENESIS, maybe we should set the default value to `pwd` in configure.ac so that both GENESIS' build variants behave the same.

The alternative is to enforce the --prefix parameter (taken from [NEST](http://nest-initiative.org/Software:Download)):

```
# If the user did not supply --prefix, we fail loudly
if test $prefix = NONE; then
  echo "Please supply a valid installation prefix, for example ./configure --prefix=\$HOME/opt !"
  exit 1
fi
```

What do you think?
Also, should we include the version name in the default INSTALLDIR for autoconf? Right now, it's `INSTALLDIR=@prefix@/genesis` but we could change that to something like `INSTALLDIR=@prefix@/genesis-$(VERSION)`.

**Comment by dbeeman** [comment]
I would prefer to keep consistency with the conventions used in previous releases, as much as possible, so as not to confuse present GENESIS users.  Do you see any problem with sticking with 'pwd'? Including the version name in INSTSTALLDIR sounds like a good idea.

**Comment by physicalist** [comment]
We could do that I guess. In pull request #24, 884c4b5b8724558be3e6711c1fde46fef9871b2d implements rather the _enforce active user choice_ method as described above, which has the advantage that users immediately are made aware that they need to decide where to put the binaries when `make install`ing. No need to fiddle with the Makefiles to even find out how they would change the install directory.

In the follow-up commit cf383b0758bb699180b6c45ac5c8b698dfbb9cce I adapted both the configure script and [Makefile.dist](../blob/master/src/Makefile.dist) so that it uses the version in the install directory. In both cases on can still directly override the installation path by calling `INSTALLDIR=<path> make install`. In the latter (Makefile.dist) case, I added the option to call `PREFIX=<path> make install`, in which case genesis would be installed to `<path>/genesis-2.4` (the version is defined at the very top of Makefile.dist for convenient changing afterwards & affects VERSNAME in the bindist target further down).

@dbeeman If you agree to these changes, feel free to apply both of them.

**Comment by dbeeman** [comment]
Still a problem.  INSTALLDIR appends "genesis-2.4"  to the prefix, but it should add 'genesis',
so that  "./config --prefix=/usr/local/genesis-2.4" will install in /usr/local/genesis-2.4/genesis.

**Comment by dbeeman** [comment]
Actually, it would make more sense to set INSTALLDIR literally to the prefix string.  Then I would specify --prefix=/usr/local/genesis-2.4/genesis as exactly where i want the 'genesis' directory to be.
This is essentialy what 'pwd/..'  did before.

**Comment by physicalist** [comment]
Well, if that's what you prefer, the following patch would do that. Keep in mind though that this is not the expected behaviour of `./configure --prefix=<path>`; people who are used to the autoconf way of installing would surely be confused by this. I think it is rather an issue of documentation, i.e. describe in the README that one can change the path after configure when installing with `INSTALLDIR=<path> make install`.
For clarification, I added some more text to the configure script that's displayed when ./configure is run without --prefix.

If you prefer to default to `..` still, you change src/configure.ac in that you remove this block:

```
# If the user did not supply --prefix, we fail loudly
# (taken from NEST's configure.ac)
if test $prefix = NONE; then
  echo "Please supply a valid installation prefix!"
  echo "GENESIS will then be installed into \$PREFIX/genesis-2.4"
  echo "Example: to install GENESIS into \$HOME/opt/genesis-2.4, run the following:"
  echo "./configure --prefix=\$HOME/opt"
  exit 1
fi
```

(or just the line `exit 1`)
and replace it with `AC_PREFIX_DEFAULT(..)` (don't forget to adjust the INSTALLDIR line in Makefile.in!)
Then run `autoconf` (no parameters!) in the src directory, autoconf will automatically generate the ./configure script from configure.ac

**Comment by dbeeman** [comment]
I think that the pactch will not be necessary.

**Comment by dbeeman** [comment]
I think that you meant change the installation path with `make install INSTALLDIR=<path>`, not the other order.  Right?

**Comment by physicalist** [comment]
Both orders work in my case but since it didn't on your system, `make install INSTALLDIR=<path>` is the way to go here.

**Comment by physicalist** [comment]
Okay I was wrong after all when it comes to how the prefix should be defined & used in `./configure --prefix=<path>`. The expected behaviour _is_ that `<path>` is the full target directory name, not that it is extended with the package name (i.e., `<path>/genesis`). I apologize, I looked at all the wrong examples :/

I patched it again in 48ece3fc754f59fec093e8903e110028853ffae7, @dbeeman please update the README accordingly. Sorry for the inconvenience.
(I would still argue that the `make install INSTALLDIR=<path>` alternative should remain part of the installation instructions in the README)

**Comment by physicalist** [comment]
Whoops, introduced a regression which caused `make install` to fail:
`INSTALLDIR ?= /path # comment` defines the variable as `INSTALLDIR="/path "` (notice the space!) which messed up creating directories & copying files & many other things. Corrected with 898b5f92c823e679513fdfdfbc3a5330ae0976b1.
